### PR TITLE
STELLAR-1997 Evaluator flowgraph for quality evaluation tool

### PR DESCRIPTION
**Pull request by reiinakano**
evaluator grc

Introduces two new blocks: Folder Source and PRBS Sink

Folder source reads a folder of binary packets and passes them to the rest of the flowgraph as PDUs. This will be used directly with the packets we collect from GCS after a test pass.

PRBS sink reads the decoded and error corrected packets and outputs information like Frame Error Rate, number of duplicates, etc.

### CODE REVIEW

**Comment by kenichi-fukushima** [comment]
Can you apply clang-format to C++ code?

**Comment by reiinakano** [comment]
Done

File: gr-starcoder_utils/include/starcoder_utils/folder_source_pdu.h [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+#define INCLUDED_STARCODER_UTILS_FOLDER_SOURCE_PDU_H
+
+#include <starcoder_utils/api.h>
+#include <gnuradio/sync_block.h>
+
+namespace gr {
+namespace starcoder_utils {
+
+/*!
+ * \brief <+description of block+>
```
Comment (kenichi-fukushima) [inline gr-starcoder_utils/include/starcoder_utils/folder_source_pdu.h]: Please document this block just as you do for prbs_sink_pdu.

Also remove comment generated by the tool for you.

File: gr-starcoder_utils/python/prbs_sink_pdu.py [hunk 2]
```diff
@@ -0,0 +1,147 @@
...
+        packet_idx = self.verify_packet_and_return_idx(arr)
+        if packet_idx == -1:
+            self.log.warn("Received erroneous packets even after FEC.")
+            self.erroneous_packets_after_fec_ctr += 1
+            return
+        self.correct_packets_received_ctr += 1
+        self.collected_packets[packet_idx] += 1
+
+    def verify_packet_and_return_idx(self, arr):
+        packet_idx = arr[3] << 24
```
Comment (kenichi-fukushima) [inline gr-starcoder_utils/python/prbs_sink_pdu.py]: Comment this chunk of code or factor it out as a function to give it name.

File: gr-starcoder_utils/python/prbs_sink_pdu.py [hunk 3]
```diff
@@ -0,0 +1,147 @@
...
+        self.statistics = None
+
+        self.message_port_register_in(pmt.intern("all"))
+        self.message_port_register_in(pmt.intern("corrected"))
+
+        self.set_msg_handler(pmt.intern("all"), self.all_handler)
+        self.set_msg_handler(pmt.intern("corrected"), self.corrected_handler)
+
+    def all_handler(self, msg):
+        if not pmt.is_pair(msg):
```
Comment (kenichi-fukushima) [inline gr-starcoder_utils/python/prbs_sink_pdu.py]: I see this ladder of pmt type checks so many time. You can create a utility method for it (in another PR)

**Comment by reiinakano** [comment]
Done

**Pull request merged**